cranelift/codegen/src/isa/s390x/inst.isle

```diff
@@ -3453,26 +3419,22 @@
 ;; if that unrelated instruction does not modify the condition code.)
 (decl emit_producer (ProducesFlags) Unit)
 (rule (emit_producer (ProducesFlags.ProducesFlagsSideEffect insn)) (emit insn))
-(decl emit_consumer (ConsumesFlags) Unit)
-(rule (emit_consumer (ConsumesFlags.ConsumesFlagsReturnsReg insn _)) (emit insn))
 
 ;; Use a boolean condition to select between two registers.
 (decl select_bool_reg (Type ProducesBool Reg Reg) Reg)
 (rule (select_bool_reg ty (ProducesBool.ProducesBool producer cond) reg_true reg_false)
-      (let ((dst WritableReg (temp_writable_reg ty))
-            (_ Unit (emit_producer producer))
-            (_ Unit (emit_mov ty dst reg_false))
-            (_ Unit (emit_consumer (emit_cmov_reg ty dst cond reg_true))))
-        dst))
+      (let ((dst WritableReg (temp_writable_reg ty)))
+        (with_flags_reg
+          producer
+          (emit_cmov_reg ty dst cond reg_false reg_true))))
 
 ;; Use a boolean condition to select between two immediate values.
 (decl select_bool_imm (Type ProducesBool i16 u64) Reg)
 (rule (select_bool_imm ty (ProducesBool.ProducesBool producer cond) imm_true imm_false)
-      (let ((dst WritableReg (temp_writable_reg ty))
-            (_ Unit (emit_producer producer))
-            (_ Unit (emit_imm ty dst imm_false))
-            (_ Unit (emit_consumer (emit_cmov_imm ty dst cond imm_true))))
-        dst))
+      (let ((dst WritableReg (temp_writable_reg ty)))
+        (with_flags_reg
+          producer
+          (emit_cmov_imm ty dst cond imm_true (imm ty imm_false)))))
```

Comment:
So the old `select_bool_imm/reg` were written the way they were specifically to avoid unnecessary register moves.  It appears this new way brings those unnecessary moves back (see e.g. the many tests with an extra `lgr` in `icmp.clif`.  This is not a show-stopper for this PR, we can look into it afterwards -- just wondering if there's something we can to do improve on this again ...

Reply:
I went ahead and reverted this commit, as it's not strictly necessary for removing uses of `reg_mod`.